File: baekjoon/p07770/Main.java
package baekjoon.p07770;

import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        System.out.println(solve(n));
    }

    private static int solve(int n) {
        int height = 0;
        int block = 1;
        int add = 4;
        int under = 1;

        while (block <= n) {
            height++;
            under += add;
            add += 4;
            block += under;
        }

        return height;
    }
}

File: baekjoon/p07770/Main.kt
package baekjoon.p07770

fun main() {
    val n = readLine()!!.toInt()
    println(solve(n))
}

private fun solve(n: Int): Int {
    var height = 0
    var block = 1
    var under = 1
    var add = 4

    while (block <= n) {
        height++
        under += add
        add += 4
        block += under
    }

    return height;
}